R/footnote_plan.R

```diff
@@ -0,0 +1,96 @@
+#' Footnote Plan
+#'
+#' @param ...  footnote structure objects separated by commas
+#' @param marks type of marks required for footnotes. available options are "numbers", "letters", "standard" and "extended".
+#' The default option is set to "numeric".
+#'
+#' @return footnote plan object
+#' @export
+#'
+#' @examples
+#'
+#' footnote_plan <- footnote_plan(
+#'     footnote_structure(footnote_text = "Source Note"),
+#'     marks="letters")
+#'
+#' footnote_plan <- footnote_plan(
+#'     footnote_structure(footnote_text = "footnote", column_val = "Placebo"),
+#'     marks="numbers")
+#'
+footnote_plan <- function(...,marks=c("numbers","letters","standard","extended")){
+  footnote_structure_list <- list(...)
+  marks = match.arg(marks)
+
+  structure(
+    list(struct_list=footnote_structure_list, marks=marks),
+    class = c("footnote_plan", "plan")
+  )
+}
+
+#' Footnote Structure
+#'
+#' @param footnote_text string with text for footnote
+#' @param column_val string or a named list of strings which represent the column to apply the footnote to
+#' @param group_val string or a named list of strings which represent the value of group to apply the footnote to
+#' @param label_val string which represents the value of label to apply the footnote to
+#'
+#' @return footnote structure object
+#' @export
+#'
+#' @examples
+#'
+#' footnote_structure <- footnote_structure(footnote_text = "Source Note")
+#'
+#' footnote_structure <- footnote_structure(footnote_text = "Text",
+#'                                column_val = "Placebo")
+#'
+#' footnote_structure <- footnote_structure(footnote_text = "Text",
+#'       column_val = list(col1 = "Placebo", col2= "Treatment groups"))
+#'
+#' footnote_structure <- footnote_structure("Text",label_val = "Adverse Event")
+footnote_structure <- function(footnote_text, column_val = NULL, group_val = NULL, label_val = NULL){
+
+  # force column_val and group_val to list
+  if(length(column_val)>1 && is.list(column_val)==FALSE){
+    column_val <- as.list(column_val)
+  }else if(length(column_val)==1 && !is.null(names(column_val))){
+    column_val<-as.list(column_val)
+  }
+
+  if(length(group_val)>1 && is.list(group_val)==FALSE){
+    group_val <- as.list(group_val)
+  }else if(length(group_val)==1 && !is.null(names(group_val))){
+    group_val<-as.list(group_val)
+  }
+
+  # warnings if elements arent named
+
+  if(is.list(column_val)){
+    column_val_names <- names(column_val)
+    if(is.null(column_val_names)){
+      stop("when column_val is a list, must be a named list")
```

Comment:
It looks like this will throw an error if a user supplies a vector of values (e.g. if they have 1 column variable and want to add the footnote to several column labels). Is that expected?

Reply:
So if you had several columns at the same level you wanted to add a footnote to you say 
```
column_val = list(foo= c("a", "b", "c"))
```